### PR TITLE
Docs: add dependency installation commands for RHEL based systems

### PR DESCRIPTION
**Pull request by Stef16Robbe**
### Changes
- Adds local installation instructions for RHEL based machines in `docs/source/technical_design/localinstall.rst`
- It also updates the NodeJS commands to not use the [deprecated installation scripts](https://github.com/nodesource/distributions#new-update-%EF%B8%8F) (this should also be done for Debian systems but I have not spent the time on that yet)
  - Yarn installation is done according to: https://classic.yarnpkg.com/lang/en/docs/install/#centos-stable
  - Tested on Fedora 39
- Also fixes an issue with inproper URL link syntax towards Docker docs

### Issue link
Closes [2058](https://github.com/minvws/nl-kat-coordination/issues/2058)

### Demo
N.A.

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by underdarknl — DISMISSED** [review]

**Comment by underdarknl** [comment]
@Stef16Robbe thank you! RHEL should be a first class citizen in the OpenKat ecosystem.
It looks like our precommit tool does not like the empty new line at the end of the docs though, could you remove that?

File: docs/source/technical_design/localinstall.rst [hunk 1]
```diff
@@ -126,4 +138,5 @@ OpenTelemetry
 
 OpenTelemetry is a way to trace requests through the system. It is used to find out where a request is going wrong and to instrument performance problems. OpenTelemetry is not enabled by default, but can be enabled by uncommenting the environment variable ``SPAN_EXPORT_GRPC_ENDPOINT`` in the ``.env`` file.
 
-The `Jaeger <https://www.jaegertracing.io>`_ tracing system is used to view the traces. It can be enabled by enabling the `Docker Compose profile <https://docs.docker.com/compose/profiles/#enable-profiles>`, for example by running ``docker-compose --profile jaeger up -d`` or using ``export COMPOSE_PROFILES=jaeger`` and then running Make as usual. The Jaeger UI can then be found at http://localhost:16686.
+The `Jaeger <https://www.jaegertracing.io>`_ tracing system is used to view the traces. It can be enabled by enabling the `Docker Compose profile <https://docs.docker.com/compose/profiles/#enable-profiles>`_, for example by running ``docker-compose --profile jaeger up -d`` or using ``export COMPOSE_PROFILES=jaeger`` and then running Make as usual. The Jaeger UI can then be found at http://localhost:16686.
+
```
Comment (Rieven) [inline docs/source/technical_design/localinstall.rst:142]:
```suggestion
```

**Review by ammar92 — APPROVED** [review]

**Comment by Stef16Robbe** [comment]
> @Stef16Robbe Thanks! This is a nice addition to our docs. I'm currently trying to get it to work on Fedora. During the build phase it has troubles connecting and building the Postgres database. Haven't figured out yet what is going wrong, so I'll continue debugging.

@stephanie0x00 Good thing you double checked! On my current working install I also get some Postgres "Connection refused" logs, but it does work after a few tries:

```
while ! nc -vz $ROCKY_DB_HOST $ROCKY_DB_PORT; do sleep 0.1; done
nc: connect to postgres (172.20.0.2) port 5432 (tcp) failed: Connection refused
...
nc: connect to postgres (172.20.0.2) port 5432 (tcp) failed: Connection refused
Connection to postgres (172.20.0.2) 5432 port [tcp/postgresql] succeeded!
```

I did a sanity check on a completely clean install again, and I seem to run into the same issue as you are:

```
while ! nc -vz $ROCKY_DB_HOST $ROCKY_DB_PORT; do sleep 0.1; done
nc: connect to postgres (172.20.0.2) port 5432 (tcp) failed: Connection refused
nc: connect to postgres (172.20.0.2) port 5432 (tcp) failed: Connection refused
nc: connect to postgres (172.20.0.2) port 5432 (tcp) failed: Connection refused
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
nc: getaddrinfo for host "postgres" port 5432: Name or service not known
...
```

On my working system I haven't done any installs related to postgres (to my knowledge, I checked my dnf history), so I'm not sure where this is coming from.

First thoughts were [installing postgresql](https://docs.fedoraproject.org/en-US/quick-docs/postgresql/), and initializing the database using `sudo /usr/bin/postgresql-setup --initdb`

nc is looking for the postgres server so I thought I should start it: `sudo systemctl enable --now postgresql`

But `make kat` didn't like that:

```
...
Creating nl-kat-coordination_postgres_1 ... 
Creating nl-kat-coordination_postgres_1 ... error

ERROR: for nl-kat-coordination_postgres_1  Cannot start service postgres: driver failed programming external connectivity on endpoint nl-kat-coordination_postgres_1 (7d1b99cc06624ead1877e2299978e00ca801ac15354bbfa2d4144c6d74d36a1a): Error starting userland proxy: listen tcp4 127.0.0.1:5432: bind: address already in use

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint nl-kat-coordination_postgres_1 (7d1b99cc06624ead1877e2299978e00ca801ac15354bbfa2d4144c6d74d36a1a): Error starting userland proxy: listen tcp4 127.0.0.1:5432: bind: address already in use
ERROR: Encountered errors while bringing up the project.
...
```

I'm not familiar enough yet with OpenKat's build process to solve this immediately but I'll look into it.

**Comment by Stef16Robbe** [comment]
The error says it all basically, and it can also be seen in the `docker logs` output I shared. `POSTGRES_PASSWORD` and other secrets were empty because I hadn't ran `make env` yet. After that everything ran smoothly and I was able to start OpenKat and walk through the onboarding.

`make env` is described in `docs/source/technical_design/containers.md`. Shouldn't that also be included here?

**Comment by stephanie0x00** [comment]
@Stef16Robbe The installation of `openssl` seemed to be the trick indeed. I don't recall that running `make env` was necessary on Debian/Ubuntu. I'm in favour of adding this to the documentation for all systems. 

What I also noticed was that our current installation steps are a bit confusing as to whether or not to install Docker from the repositories, as we mention that `docker-compose` should be installed. We advise to install via Docker directly, as the debian/ubuntu repos are offering an outdated docker which doesn't have some of the features that are being required for OpenKAT. This might also be the case for (certain) Fedora versions.

**Review by stephanie0x00 — DISMISSED** [review]
QA on Fedora with the install commands works. Ready for merge.

**Review by underdarknl — APPROVED** [review]

**Pull request merged**